react/Pill/Pill.js

```diff
@@ -28,10 +29,11 @@ export default class Pill extends Component<Props> {
 
   renderStaticPill() {
     const { children, text, className, ...restProps } = this.props;
+    const footerStyle = this.props.footer ? styles.footerBadge : null;
```

Comment:
The header/footer badge styles are non-standard compared to the rest of the system, so I don't think we should make this an option of `Pill`. If you hard-code this style inside of `Footer`, that should be fine.